pkg/controller/wildflyserver/wildflyserver_controller.go

```diff
@@ -130,12 +170,161 @@ func (r *ReconcileWildFlyServer) Reconcile(request reconcile.Request) (reconcile
+		wildflyServerSpecPodStatus := getWildflyServerPodStatusByName(wildflyServer, scaleDownPodName)
+		scaleDownPodsState[scaleDownPodName] = wildflyServerSpecPodStatus.State
+		if wildflyServerSpecPodStatus.State != wildflyv1alpha1.PodStateScalingDownClean {
+			// When state is not already processed with recovery and all data was cleaned then we mark pod to be dirty
+			scaleDownPodsState[scaleDownPodName] = wildflyv1alpha1.PodStateScalingDownDirty
+		}
+
+		if wildflyServerSpecPodStatus.State != wildflyv1alpha1.PodStateScalingDownClean {
+			// Removing the pod from the Service handling
+			scaleDownPod.ObjectMeta.Labels[labelWildflyOperatorInService] = "under-scale-down-processing"
+			if err = r.client.Update(context.TODO(), &scaleDownPod); err != nil {
+				reqLogger.Error(err, "Failed to update pod labels", "Pod name", scaleDownPod.ObjectMeta.Name, "Pod labels ", scaleDownPod.ObjectMeta.Labels)
+				scaleDownErrors = append(scaleDownErrors, err)
+				break
+			}
+
+			reqLogger.Info("Going for scaledown with pod", "pod name", scaleDownPod.ObjectMeta.Name, "IP address", scaleDownPodIP)
+
+			managementURL := fmt.Sprintf("http://%v:%v/management", scaleDownPodIP, httpManagementPort)
+			username := string(foundTxnRecoverySecret.Data["username"])
+			password := string(foundTxnRecoverySecret.Data["password"])
+			httpDigest := wildflyutil.HTTPDigest{URL: managementURL, Username: username, Password: password}
+
+			// Enabling recovery listener to speed up recovery processing
+			jsonResult, err := wildflyutil.ExecuteMgmtOp(&httpDigest, wildflyutil.MgmtOpTxnEnableRecoveryListener)
+			if err != nil {
+				reqLogger.Error(err, "Cannot enable transaction recovery listener", "Management URL", managementURL, "Pod name", scaleDownPod)
+				scaleDownErrors = append(scaleDownErrors, err)
+				break
+
+			}
+			if !wildflyutil.IsMgmtOutcomeSuccesful(jsonResult) {
+				reqLogger.Info("Failed to enable transaction recovery listener. Scaledown processing can take longer",
+					"Management command", wildflyutil.MgmtOpTxnEnableRecoveryListener, "JSON response", jsonResult)
+			}
+			// Enabling the recovery listner may require the server being reloaded
+			isReloadRequired := wildflyutil.ReadJSONDataByIndex(jsonResult["response-headers"], "operation-requires-reload")
+			if isReloadRequiredBool, _ := strconv.ParseBool(isReloadRequired); isReloadRequiredBool {
+				jsonResult, err = wildflyutil.ExecuteMgmtOp(&httpDigest, wildflyutil.MgmtOpReload)
+				if err != nil {
+					reqLogger.Error(err, "Cannot reload application container", "Management URL", managementURL, "Pod name", scaleDownPod)
+					scaleDownErrors = append(scaleDownErrors, err)
+					break
+				}
+				for serverStateCheckCounter := 0; err != nil && serverStateCheckCounter < reloadRetryCount; serverStateCheckCounter++ {
+					jsonResult, err = wildflyutil.ExecuteMgmtOp(&httpDigest, wildflyutil.MgmtOpServerStateRead)
```

Comment:
The operator is blocking until the app server's pod is reloaded, right?

That'd not be correct, the reconcile loop should not be blocking like this.
You would have to issue a :reload and then return `reconcile.Result{Requeue: true}, nil` so that the reconcile loop can handle other `WildFlyServer` events
Once you go back to your own `WildFlyServer` resource, you know from the pod state that it was reloading and then the code can proceed further